client/src/js/actions/user.js

```diff
@@ -0,0 +1,87 @@
+import { updateMessages } from './messages';
+
+export const RESET_USER = 'RESET_USER';
+export function resetUser() {
+  return (dispatch, __, socket) => {
+    socket.jwt = null;
+    dispatch({ type: RESET_USER });
+  }
+}
+
+export const UPDATE_USER = 'UPDATE_USER';
+export function updateUser(update) {
+  return {
+    type: UPDATE_USER,
+    update
+  };
+}
+
+// This action connects to the websocket server, but does NOT authenticate.
+export function connectUser() {
+  return async (dispatch, _, socket) => {
+    try {
+      
+      // Create thunk that will update redux state when users socket becomes disconnected.
+      const onDisconnect = () => dispatch(updateUser({ connected: false }));
+
+      // All users should get message updates as a firehose once connected to the wss
+      socket.on('currentMessageState', (messages) => {
+        dispatch(updateMessages(messages))
+      });
+
+      socket.on('error', (error) => {
+        console.error(error);
+        // TODO: Display this in UI somewhere.
+      });
+
+      await socket.connect('ws://localhost:3000', onDisconnect);
+
+      // At this point we are connected, but not authenticated or associated with a user account.
+      // The user will need to either log in or sign up to authenticate.
+      dispatch(updateUser({ connected: true }));
+    
+      return;
```

Comment:
@alexander-daniel Why is a return necessary here?

Reply:
It is not!